### ISSUE TITLE
Support LineageOS

### ISSUE THREAD

**Issue by protist** [ScreamingHawk/phone-saver#30]
Support request. Generated by Phone Saver.

Intent type: text/plain
Intent action: android.intent.action.SEND
Text: http://example.com/
Subject: Example Domain
Application Version: 2.2

More information:
As per issue #19, this share was generated by navigating to http://example.com in Android Firefox, then sharing with Phone Saver. I'm fairly sure it's just sharing plain text, being the url. When I "share" this page with a file browser instead, it just saves a plain text file that contains the url.

**Comment by ScreamingHawk** [comment]
>When I "share" this page with a file browser instead, it just saves a plain text file that contains the url.

Sounds easy

**Comment by ScreamingHawk** [comment]
Live in v2.5 (currently in the beta stream on Google Play)

**Comment by protist** [comment]
Thank you! I'll check it out and report back after it hits F-droid. Cheers.

**Comment by protist** [comment]
I just installed v2.5 from F-droid, but it came up again as `Unsupported Content Type`.

**Comment by ScreamingHawk** [comment]
Hmm. Me too. Weird because it was working on my earlier build. I'll look into it again.

**Comment by ScreamingHawk** [comment]
Found the problem. The `charset` is being passed in through with the `content-type` which breaks my test for `text/html`. 

Fixed for next release: v2.6+

**Comment by ScreamingHawk** [comment]
Feel free to reopen this issue if you see any problems :)

**Comment by protist** [comment]
Hi, I just upgraded to v3.0 from F-droid, but Phone Saver still says `Unsupported Content Type`.

(Also, I can't reopen this issue because it wasn't me that closed it.)

**Comment by ScreamingHawk** [comment]
It's working for me in Chrome on the couple I sites I had open. 
And from Firefox on example.com. 

Which site in particular are you trying to save?

**Comment by protist** [comment]
Odd, I'm testing with Firefox (latest, 54.0.1) and example.com too. Raising a feature request through Phone Saver comes up with the same information as the OP.

> Support request. Generated by Phone Saver.
> 
> Intent type: text/plain
> Intent action: android.intent.action.SEND
> Text: http://example.com/
> Subject: Example Domain
> Application Version: 3.0

**Comment by ScreamingHawk** [comment]
Do you see a `URL Content-Type` (not just `Content-Type`) appear in the generated ticket? 
You may have to look in the URL if it doesn't show in the GitHub box automatically.

**Comment by protist** [comment]
No, I don't think so. Here is the url produced by Phone Saver:

> https://github.com/ScreamingHawk/phone-saver/issues/new?title=Support%20Request%20-%20text/plain&body=Support%20request.%20Generated%20by%20Phone%20Saver.%0D%0A%0D%0AIntent%20type:%20text/plain%0D%0AIntent%20action:%20android.intent.action.SEND%0D%0AText:%20http://example.com/%0D%0ASubject:%20Example%20Domain%0D%0AApplication%20Version:%203.0%0D%0A%0D%0AMore%20information:%20TYPE_ADDITIONAL_INFORMATION_HERE%0D%0A%0D%0AThank%20you

**Comment by protist** [comment]
For the record, I just updated from 3.0 to 3.2, and this issue is still present. I tested the new `Force saving` function, but when attempting to save, it just says `Save failed`.

**Comment by ScreamingHawk** [comment]
That's really odd. I still haven't been able to replicate this issue with Firefox. 

I'll make a point to add some searching logging to help debug this.

**Comment by ScreamingHawk** [comment]
I'm renaming this ticket for Firefox support, since that seems to be the real issue.

**Comment by ScreamingHawk** [comment]
#48 for adding user visible logging

**Comment by ScreamingHawk** [comment]
I've added some more mime type validation. I originally suspected this was the issue with Firefox... It may or may not help you :/ Still looking into the best way to implement #48 to actually debug your problem.

**Comment by protist** [comment]
> I've added some more mime type validation.

I'm not sure if you are referring to version 3.4, but the issue still persists for me.

> I'm renaming this ticket for Firefox support, since that seems to be the real issue.

I don't think it's just Firefox on my system. I tried sharing from the bundled "Browser" in LineageOS, and again got `Save failed`.

**Comment by ScreamingHawk** [comment]
Perhaps LineageOS is handling the share feature differently then. 

I'll rename this ticket again, will have to wait for #48 to get more information to debug this sorry.

**Comment by ScreamingHawk** [comment]
I've updated the build tools, which has fixed a couple of issues around LineageOS. #53 Doesn't seem related but may have fixed this issue. I'm going to have a look at adding more user notifications too just in case.

**Comment by protist** [comment]
I just tested v3.8 (this time from Google Play), and I again got `Unsupported Content Type`.

Honestly, I had forgotten about this, as I had since moved on to other solutions. I strongly doubt that I'd be using Phone Saver in the future, since I've another solution now. Hence, if I was the only one bitten by this issue, then I'm happy for it to be closed, or left open for reference, or whichever you prefer. Thank you for your support.
